### PR TITLE
newsraft: init at 0.21

### PR DESCRIPTION
**Pull request by arthsmn**
## Description of changes

https://codeberg.org/grisha/newsraft

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix
Comment (IogaMaster) [inline pkgs/top-level/all-packages.nix]:
```suggestion
```

File: maintainers/maintainer-list.nix
Comment (IogaMaster) [inline maintainers/maintainer-list.nix:1533]: Add this in a separate commit titled

`maintainers: add arthsmn`

**Review by IogaMaster — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ne/newsraft/package.nix
Comment (IogaMaster) [inline pkgs/by-name/ne/newsraft/package.nix:5]:
```suggestion
, curl
, expat
, gumbo
, ncurses
, sqlite
, yajl
```

**Review by IogaMaster — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ne/newsraft/package.nix
Comment (IogaMaster) [reply pkgs/by-name/ne/newsraft/package.nix:5]: These are minor nitpicks, but I feel you should know the general style of nixpkgs contributions, considering this is your first pr.

**Comment by Steinhagen** [comment]
Also something related to the nixpkgs PRs that I've seen is that each should contain a single commit with the good convention name.
I don't think people are squashing the commits before merging.

**Comment by IogaMaster** [comment]
Make the commits into this history

- maintainers: add arthsmn
- newsraft: init at 0.21

**Review by IogaMaster — APPROVED** [review]
Looks great!
Welcome to nixpkgs and thanks for contributing! :smile:

**Comment by Steinhagen** [comment]
Looks good. Gave it a try and worked as expected. Nice one!

**Comment by IogaMaster** [comment]
@Steinhagen would you go to the files section and add a review approval if you can. This will help the pr get merged. Thanks for your help!

**Review by Steinhagen — APPROVED** [review]
LGTM!

**Comment by IogaMaster** [comment]
Thank you Anderson!

**Comment by ajwhouse** [comment]
@arthsmn Yo, thanks for adding this! I've had it on my list for weeks to add this one next, but here I see you already did. Cheers.

**Pull request merged**
